app/controllers/application_controller.rb

```diff
@@ -53,6 +53,13 @@ def redirect_to_root
     redirect_to root_path
   end
 
+  def verify_with_otp
+    return unless @api_user.mfa_login_and_write?
+    otp = request.headers["HTTP_OTP"] || ''
+    return if @api_user&.otp_verified?(otp)
```

Comment:
is this `&` needed here? `verify_authenticated_user` ensures that user see 401 if `@api_user` is nil.